crates/linker/src/lib.rs

```diff
@@ -110,9 +107,9 @@ pub fn build_and_preprocess_host(
 
     preprocess(
         target,
-        dynhost.to_str().unwrap(),
-        metadata.to_str().unwrap(),
-        preprocessed_host_path.to_str().unwrap(),
+        &dynhost,
+        &metadata,
+        preprocessed_host_path,
```

Comment:
not sure why paths were turned into strings here?! Path captures the usage better I think

Reply:
No particular reason. Your change sounds better.